src/Extensions/Logging/Internal/Replayable/ActivityExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Omex.Extensions.Logging.Replayable;
+
+namespace Microsoft.Omex.Extensions.Logging.Internal.Replayable
+{
+	internal static class ReplayebleActivityExtensions
+	{
+		private const string ReplayableLogKey = "ReplayableLogKey";
+
+		private static ConcurrentQueue<LogMessageInformation>? GetReplayableLogsInternal(this Activity activity) =>
+			activity.GetCustomProperty(ReplayableLogKey) as ConcurrentQueue<LogMessageInformation>;
+
+		public static IEnumerable<LogMessageInformation> GetReplayableLogs(this Activity activity) =>
+			activity.GetReplayableLogsInternal() ?? Enumerable.Empty<LogMessageInformation>();
+
+		public static void AddReplayLog(this Activity activity, LogMessageInformation logEvent, uint maxReplayedEventsPerActivity)
+		{
+			ConcurrentQueue<LogMessageInformation>? queue = activity.GetReplayableLogsInternal();
+			if (queue == null)
+			{
+				queue = new ConcurrentQueue<LogMessageInformation>();
+				activity.SetCustomProperty(ReplayableLogKey, queue);
+			}
+
```

Comment:
Is this thread safe? Should we not use Interlocked.CompareExchange?

Reply:
Good point interlocker would be useful here, thank you

Reply:
After some more thinking I've realized that queue is a local variable so `Interlocker` won't help here.
More discussion of alternatives is here:  https://github.com/microsoft/Omex/pull/285#discussion_r561991954